comms/src/peer_manager/peer.rs

```diff
@@ -70,6 +73,23 @@ impl Peer {
         }
     }
 
+    /// Constructs a new peer
+    pub fn new_from_public_key_and_address(
```

Comment:
nit: Think it's idiomatic to name this function `from_public_key_and_address`.

Reply:
Done